src/glim_ros/glim_ros.cpp

```diff
@@ -37,6 +37,9 @@
 
 namespace glim {
 
+static bool init_qos_profile(const std::string& name, rmw_qos_profile_t& profile);
+static void get_qos(std::shared_ptr<spdlog::logger> logger, const Config& config_ros, const std::string& module_name, rclcpp::QoS& profile);
```

Comment:
It would be good to let `get_pos` return the QoS profile rather than taking it as an argument.